From f06084a007fdcaad3bd68f29e243fa8edcf0b590 Mon Sep 17 00:00:00 2001
From: "Alan T. DeKok" <aland@freeradius.org>
Date: Thu, 24 Dec 2020 07:38:59 -0500
Subject: [PATCH] remove unused argument.  Fixes #3817

---
 src/tests/map/map_unit.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/tests/map/map_unit.c b/src/tests/map/map_unit.c
index af6d016..df906a5 100644
--- a/src/tests/map/map_unit.c
+++ b/src/tests/map/map_unit.c
@@ -81,7 +81,6 @@ static void NEVER_RETURNS usage(void)
 	fprintf(stderr, "usage: map_unit [OPTS] filename ...\n");
 	fprintf(stderr, "  -d <raddb>             Set user dictionary directory (defaults to " RADDBDIR ").\n");
 	fprintf(stderr, "  -D <dictdir>           Set main dictionary directory (defaults to " DICTDIR ").\n");
-	fprintf(stderr, "  -O <output_dir>	  Set output directory\n");
 	fprintf(stderr, "  -x                     Debugging mode.\n");
 	fprintf(stderr, "  -M                     Show program version information.\n");
 
-- 
2.23.0

